packages/generator-adp/src/app/extension-project/index.ts

```diff
@@ -0,0 +1,86 @@
+import { resolve } from 'path';
+
+import {
+    getFormattedVersion,
+    type AttributesAnswers,
+    type ConfigAnswers,
+    type SystemLookup
+} from '@sap-ux/adp-tooling';
+
+import type { ExtensionProjectData } from '../types';
+
+export const EXTENSIBILITY_GENERATOR_NS = '@bas-dev/generator-extensibility-sub/generators/app';
+
+/**
+ * Prepares data required for generating an extension project.
+ *
+ * @param {ConfigAnswers} configAnswers - The configuration answers including system, user, and app selection.
+ * @param {AttributesAnswers} attributeAnswers - The basic project attributes like name, namespace, version.
+ * @param {SystemLookup} systemLookup - The lookup service for system destination info.
+ * @returns {Promise<ExtensionProjectData>} A promise resolving to the prepared extension project data object.
+ */
+export async function getExtensionProjectData(
+    configAnswers: ConfigAnswers,
+    attributeAnswers: AttributesAnswers,
+    systemLookup: SystemLookup
+): Promise<ExtensionProjectData> {
+    const { application, system, username, password } = configAnswers;
+    if (!application) {
+        throw new Error('Application parameters are missing.');
+    }
+
+    const destinationInfo = await systemLookup.getSystemByName(system);
+    if (!destinationInfo) {
+        throw new Error('Destination info is missing.');
```

Comment:
same as above

Reply:
Changed in [ea4f106](https://github.com/SAP/open-ux-tools/pull/3146/commits/ea4f106cb85bcb9ac9ece36c041ed870b34617aa)